elide-graphql/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -563,22 +562,38 @@ private ConnectionContainer replaceObjects(Environment context) {
         return sort.map(Sorting::parseSortRule);
     }
 
+    private MultivaluedHashMap<String, String> getQueryParams(Optional<String> typeName, String filterStr) {
+        return new MultivaluedHashMap<String, String>() {
+            {
+                String filterKey = "filter";
+                if (typeName.isPresent()) {
+                    filterKey += "[" + typeName + "]";
+                }
+                put(filterKey, Arrays.asList(filterStr));
+            }
+        };
+    }
+
     private Optional<FilterExpression> buildFilter(String typeName,
                                                    Optional<String> filter,
                                                    RequestScope requestScope) {
         // TODO: Refactor FilterDialect interfaces to accept string or List<String> instead of (or in addition to?)
         // query params.
         return filter.map(filterStr -> {
-            MultivaluedMap<String, String> queryParams = new MultivaluedHashMap<String, String>() {
-                {
-                    put("filter[" + typeName + "]", Arrays.asList(filterStr));
-                }
-            };
+            String errorMessage = "";
+            try {
+                return requestScope.getFilterDialect()
+                        .parseGlobalExpression(typeName, getQueryParams(Optional.empty(), filterStr));
+            } catch (ParseException e) {
+                errorMessage = e.getMessage();
+            }
+
             try {
-                return requestScope.getFilterDialect().parseTypedExpression(typeName, queryParams).get(typeName);
+                return requestScope.getFilterDialect()
+                        .parseTypedExpression(typeName, getQueryParams(Optional.of(typeName), filterStr))
+                        .get(typeName);
             } catch (ParseException e) {
-                log.debug("Filter parse exception caught", e);
-                throw new InvalidPredicateException("Could not parse filter for type: " + typeName);
+                throw new InvalidPredicateException(errorMessage + "\n" + e.getMessage());
```

Comment:
@aklish will this conflict with #1304?

Reply:
It will - but should be simple to merge.